packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -336,12 +350,39 @@ internal final class FileDownloader: NSObject, URLSessionDataDelegate {
 
   func parseManifestResponse(
     _ httpResponse: HTTPURLResponse,
-    withData data: Data,
+    withData data: Data?,
     database: UpdatesDatabase,
     successBlock: @escaping RemoteUpdateDownloadSuccessBlock,
     errorBlock: @escaping RemoteUpdateDownloadErrorBlock
   ) {
     let headerDictionary = httpResponse.allHeaderFields as! [String: Any]
+    let responseHeaderData = ResponseHeaderData(
+      protocolVersion: headerDictionary.optionalValue(forKey: "expo-protocol-version"),
+      serverDefinedHeadersRaw: headerDictionary.optionalValue(forKey: "expo-server-defined-headers"),
+      manifestFiltersRaw: headerDictionary.optionalValue(forKey: "expo-manifest-filters"),
+      manifestSignature: headerDictionary.optionalValue(forKey: "expo-manifest-signature")
+    )
+
+    guard let data = data else {
+      if responseHeaderData.protocolVersion == "1" && httpResponse.statusCode == 204 {
```

Comment:
Similar comment as on the Kotlin code about the protocol version check.
Also, do we care if the status code is 204, or is a 200 OK as well? (But if there's an empty 4xx or 5xx, we want to fail.) Or does an empty 200 response create a Data object that is empty? I am trying to understand when `data` is nil/null vs an empty Data instance, if ever.